cellpack/autopack/Analysis.py

```diff
@@ -927,6 +1010,335 @@ def PairCorrelationFunction_2D(self, x, y, S, rMax, dr):
+
+        if save_plots:
+            save_dir = f"{self.output_path}/heatmaps/"
+            os.makedirs(save_dir, exist_ok=True)
+
+        if get_correlations:
+            key_list = [
+                f"{pc}_{sc}"
+                for pc in range(num_packings)
+                for sc in range(num_seeds[pc])
+            ]
+            corr_df = pd.DataFrame(
+                index=key_list,
+                columns=key_list,
+                dtype=float,
+            )
+
+        for pc, packing_dict in enumerate(all_pos_list):
+            saved_plots = 0
+            for sc, (_, pos_dict) in enumerate(packing_dict.items()):
+                trial_spilr_scaled = numpy.zeros(
+                    (len(rad_vals), len(theta_vals), len(phi_vals))
+                )
+                trial_spilr_raw = copy.deepcopy(trial_spilr_scaled)
+
+                pos_list = numpy.array(pos_dict[self.ingr_key])
+                sph_pts = self.cartesian_to_sph(pos_list)
+
+                inner_loc = numpy.zeros(pos_list.shape)
+                outer_loc = numpy.zeros(pos_list.shape)
+
+                for i in range(inner_loc.shape[0]):
+                    inner_loc[i], _, _ = inner_mesh.ray.intersects_location(
+                        ray_origins=[[0, 0, 0]], ray_directions=[pos_list[i]]
+                    )
+                    outer_loc[i], _, _ = outer_mesh.ray.intersects_location(
+                        ray_origins=[[0, 0, 0]], ray_directions=[pos_list[i]]
+                    )
+
+                inner_sph_pts = self.cartesian_to_sph(inner_loc)
+                outer_sph_pts = self.cartesian_to_sph(outer_loc)
+                scaled_rad = (sph_pts[:, 0] - inner_sph_pts[:, 0]) / (
+                    outer_sph_pts[:, 0] - inner_sph_pts[:, 0]
+                )
+
+                rad_vals_raw = numpy.linspace(
+                    0, outer_sph_pts[:, 0].max(), len(rad_vals)
+                )
+
+                if (
+                    numpy.any(scaled_rad > 1)
+                    or numpy.any(scaled_rad < 0)
+                    or numpy.any(sph_pts[:, 0] > outer_sph_pts[:, 0].max())
+                    or numpy.any(sph_pts[:, 0] < 0)
+                ):
+                    raise ValueError("Check ray-mesh intersections!")
+
+                rad_inds_raw = numpy.digitize(sph_pts[:, 0], rad_vals_raw)
+                rad_inds_scaled = numpy.digitize(scaled_rad, rad_vals)
+                theta_inds = numpy.digitize(sph_pts[:, 1], theta_vals)
+                phi_inds = numpy.digitize(sph_pts[:, 2], phi_vals)
+
+                trial_spilr_scaled[rad_inds_scaled, theta_inds, phi_inds] = 1
+                trial_spilr_raw[rad_inds_raw, theta_inds, phi_inds] = 1
+
+                all_spilr_raw[pc, sc] = trial_spilr_raw.reshape((len(rad_vals_raw), -1))
+                all_spilr_scaled[pc, sc] = trial_spilr_scaled.reshape(
+                    (len(rad_vals), -1)
+                )
+
+                if saved_plots <= save_indiv_plots:
```

Comment:
This conditional is a bit confusing. I think it would help if you changed `saved_plots` to `number_of_saved_plots` and `save_indiv_plots` to something like `max_plots_to_save`? I'm not really sure what `save_indiv_plots` is supposed to be for

Reply:
Renamed these variables to clarify their purpose